src/cpp/src/scheduler.hpp

```diff
@@ -293,8 +314,11 @@ class Scheduler {
 
                 // if we can't preemt any more sequences, clear scheduled tokens and move to next sequence
                 if (!m_block_manager.can_append_slots(sequence_group)){
-                    sequence_group->clear_scheduled_tokens();
-                    continue;
+                    _try_increase_cache();
+                    if (!m_block_manager.can_append_slots(sequence_group)) {
+                        sequence_group->clear_scheduled_tokens();
+                        continue;
+                    }
```

Comment:
it seems, that we should increase memory before preemption?

Reply:
Agree, corrected.